app/src/main/java/edu/wisc/ece/pinpoint/pages/map/MapContainerFragment.java

```diff
@@ -0,0 +1,80 @@
+package edu.wisc.ece.pinpoint.pages.map;
+
+import android.Manifest;
+import android.content.Context;
+import android.location.LocationManager;
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.Toast;
+
+import androidx.activity.result.ActivityResultLauncher;
+import androidx.activity.result.contract.ActivityResultContracts;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.Fragment;
+import androidx.navigation.NavController;
+import androidx.navigation.Navigation;
+
+import com.google.android.material.floatingactionbutton.FloatingActionButton;
+
+import edu.wisc.ece.pinpoint.R;
+import edu.wisc.ece.pinpoint.pages.profile.ProfilePageFragmentDirections;
+import edu.wisc.ece.pinpoint.utils.LocationDriver;
+
+public class MapContainerFragment extends Fragment {
+    private ActivityResultLauncher<String[]> locationPermissionRequest;
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        // Code for requesting location
+        locationPermissionRequest =
+                registerForActivityResult(new ActivityResultContracts.RequestMultiplePermissions(),
+                        result -> {
+                            Boolean fineLocationGranted =
+                                    result.getOrDefault(Manifest.permission.ACCESS_FINE_LOCATION,
+                                            false);
+                            Boolean coarseLocationGranted =
+                                    result.getOrDefault(Manifest.permission.ACCESS_COARSE_LOCATION,
+                                            false);
+                            if (fineLocationGranted != null && fineLocationGranted) {
+                                // Precise location access granted.
+                            } else if (coarseLocationGranted != null && coarseLocationGranted) {
+                                // Only approximate location access granted.
+                            } else {
+                                // No location access granted.
+                            }
```

Comment:
If we don't do any conditional logic based on this info we should either comment this out or remove.